Sources/StreamChat/Controllers/ChannelController/ChannelController.swift

```diff
@@ -715,37 +718,64 @@ public class ChatChannelController: DataController, DelegateCallable, DataStoreP
         extraData: [String: RawJSON] = [:],
         completion: ((Result<MessageId, Error>) -> Void)? = nil
     ) {
-        /// Perform action only if channel is already created on backend side and have a valid `cid`.
-        guard let cid = cid, isChannelAlreadyCreated else {
-            channelModificationFailed { error in
-                completion?(.failure(error ?? ClientError.Unknown()))
-            }
-            return
-        }
-
-        /// Send stop typing event.
-        eventSender.stopTyping(in: cid, parentMessageId: nil)
-
-        updater.createNewMessage(
-            in: cid,
+        createNewMessage(
             messageId: messageId,
             text: text,
             pinning: pinning,
             isSilent: isSilent,
-            command: nil,
-            arguments: nil,
             attachments: attachments,
             mentionedUserIds: mentionedUserIds,
             quotedMessageId: quotedMessageId,
             skipPush: skipPush,
             skipEnrichUrl: skipEnrichUrl,
-            extraData: extraData
-        ) { result in
-            if let newMessage = try? result.get() {
-                self.client.eventNotificationCenter.process(NewMessagePendingEvent(message: newMessage))
-            }
-            self.callback {
-                completion?(result.map(\.id))
+            extraData: extraData,
+            poll: nil,
+            completion: completion
+        )
+    }
+    
+    /// Creates a new poll.
+    ///
+    /// - Parameters:
+    ///   - name: The name of the poll.
+    ///   - allowAnswers: Whether to allow users to provide their own answers.
+    ///   - allowUserSuggestedOptions: Whether to allow users to suggest options.
+    ///   - description: Optional description for the poll.
+    ///   - enforceUniqueVote: Whether to enforce unique votes per user.
+    ///   - maxVotesAllowed: The maximum number of votes allowed per user.
+    ///   - votingVisibility: The visibility setting for the poll.
+    ///   - options: Optional array of predefined poll options.
+    ///   - custom: Optional dictionary for any custom data associated with the poll.
+    ///   - completion: A closure to be executed once the poll is created, returning either a `MessageId` on success or an `Error` on failure.
+    public func createPoll(
+        name: String,
+        allowAnswers: Bool? = nil,
+        allowUserSuggestedOptions: Bool? = nil,
+        description: String? = nil,
+        enforceUniqueVote: Bool? = nil,
```

Comment:
The problem with this is that it is not explicit, so the default value is not transparent to the developer. We at least need to say it in the code comments. 

For example, in `ThreadListQuery` all the values are also nullable. But if I did not put the values from the backend on the SDK, then the developer would need to go to the Backend API to check what the defaults

Reply:
I documented the defaults in the method - safer to pass nil over a default value.

Reply:
IMO, I think it is actually safer to have the default values on the SDK side. Because if the customer does not pass anything, and if the backend changes the values (hopefully this does not happen) it could be a silent breaking change for the customer.

Reply:
I disagree with that - if the backend says a value is optional, then we shouldn't be making assumptions by sending default values. We already had few issues because of this, last example was in the video SDK last week.
As for your concern - the no-breaking changes policy applies for both frontend and backend, so as long as we respect the contract it's gonna be ok.

Reply:
It depends on the use case ofc. In some scenarios, making the default values explicit on the SDK is easier for the customer, because it does not require the customer to read the API documentation. Even if we add a comment saying what is the default, it can easily be out of sync with the backend. For example, queries, it is useful to see the default values in the SDK